python/pyarrow/_hdfs.pyx

```diff
@@ -25,137 +25,101 @@ from pyarrow.includes.libarrow_fs cimport *
 from pyarrow._fs cimport FileSystem
 
 
-cdef class HdfsOptions:
-    """Options for HadoopFileSystem.
+cdef class HadoopFileSystem(FileSystem):
+    """
+    HDFS backed FileSystem implementation
 
     Parameters
     ----------
-    endpoint : tuple of (host, port) pair
-        For example ('localhost', 8020).
+    host : str
+        HDFS host to connect to.
+    port : int, default 8020
+        HDFS port to connect to.
     replication : int, default 3
         Number of copies each block will have.
     buffer_size : int, default 0
         If 0, no buffering will happen otherwise the size of the temporary read
         and write buffer.
-    default_block_size : int, default None
-        None means the default configuration for HDFS, a typical block size is
+    default_block_size : int, default 0
+        0 means the default configuration for HDFS, a typical block size is
```

Comment:
I would personally keep None as the default in the user facing constructor (to me that seems clearer for a keyword that is a number where the default is not actually 0)

Reply:
Updated.

Reply:
In the signature it is still 0

Reply:
Fixed.